src/libs/actions/Report.ts

```diff
@@ -2703,12 +2703,37 @@ function updateGroupChatMemberRoles(reportID: string, accountIDList: number[], r
             onyxMethod: Onyx.METHOD.MERGE,
             key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
             value: {
-                participants,
+                participants: Object.keys(participants).reduce((acc, accountID) => {
+                    acc[Number(accountID)] = {
+                        ...participants[Number(accountID)],
+                        pendingFields: {
+                            role: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                        },
+                    };
+                    return acc;
+                }, {} as Participants),
```

Comment:
Set both `pendingAction` and `pendingFields.role` to 'update'. Don't use `reduce` but instead create two variables `optimsticParticipants` an `successParticipants`